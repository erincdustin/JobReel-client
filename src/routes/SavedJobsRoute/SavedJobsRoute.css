.saved-jobs-page {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 1em;
}

.form {
  grid-column-start: 6;
  grid-column-end: 9;
}

@media (min-width: 641px) and (max-width: 960px) {
  .form {
    grid-column-start: 1;
    grid-column-end: 9;
  }
}

@media (min-width: 320px) and (max-width: 640px) {
  .form {
    grid-column-start: 1;
    grid-column-end: 9;
  }

  .sideNavContainer {
    display: none;
  }
}

