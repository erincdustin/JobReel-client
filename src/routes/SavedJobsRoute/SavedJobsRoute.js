import React from 'react';
import SideNav from '../../components/SideNav/SideNav';
import AddJobForm from '../../components/AddJobForm/AddJobForm';
import SavedJobSummary from '../../components/SavedJobSummary/SavedJobSummary'
import jobReelApiService from '../../services/jobreel-api-service';
import JobReelContext from '../../context/JobReelContext';

class SavedJobsRoute extends React.Component {
  
  static contextType = JobReelContext

  componentDidMount = () => {
    jobReelApiService.getSavedJobs()
      .then(res => {
        this.context.setSavedJobs(res)
      })
  }

  render() {
    let display;
    (this.context.manualJobAdd === false) ? display = <SavedJobSummary /> : display = <AddJobForm /> 
    return (
      <div className="SavedJobsRoute">
        <SideNav />
        {display}
      </div>
    )
  }
}

export default SavedJobsRoute;