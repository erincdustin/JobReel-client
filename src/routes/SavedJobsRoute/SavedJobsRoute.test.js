import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import SavedJobsRoute from './SavedJobsRoute'

describe('SavedJobsRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SavedJobsRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})