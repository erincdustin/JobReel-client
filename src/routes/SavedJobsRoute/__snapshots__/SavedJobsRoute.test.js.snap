// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedJobsRoute snapshot test renders the UI as expected 1`] = `
<div
  className="saved-jobs-page"
>
  <div
    className="title"
  >
    <h2>
      Saved Jobs
    </h2>
  </div>
  <SideNav />
  <AddJobForm />
</div>
`;
